public/script.js

```diff
@@ -178,3 +181,67 @@ function notify(data, type){
     $(`#${notficationID}`).remove();
   });
 }
+
+
+
+/**
+ * Method called to show the chat message
+ * @param {object} socket The socket instance
+ * @param {object} data The data object with msg, username, room, etc
+ */
+function displayMessage(socket = null, data){
+    let { msg } = data;
+    const { user, room } = data;
+    let class_name;
+
+    if (socket.username == user) {
+      class_name = 'self';
+    } else {
+      class_name = 'others'
+    }
+
+    // Adding Emoji
+    let p;
+    let colon1 = msg.indexOf(":");
+    while (colon1 != -1) {
+      let colon2 = msg.indexOf(":", colon1 + 1);
+      if (colon2 != -1) {
+          emoji_name = msg.slice(colon1 + 1, colon2);
+          position = emoji_names.indexOf(emoji_name)
+          if (position != -1) {
+              msg = msg.slice(0, colon1) + "<img class=\"emoji\" src=\"images/emoji/" + emojis[position] + ".png\">" + msg.slice(colon2 + 1);
+          }
+          colon1 = msg.indexOf(":", colon2 + 1);
+      } else {
+          break;
+      }
+    }
+
+    // Format the current timestamp
+    const timestamp = dateFns.format(new Date(), 'H:mm:ss MMM DD');
```

Comment:
Can we format our timestamp just like the gitter does, it is very clean and there no date is displayed if the message was received today, and for all past days it displays month, date and time. See screenshots

Message sent today:
![image](https://user-images.githubusercontent.com/63332774/96566571-5d78b500-12e3-11eb-93ba-395453ae7f7f.png)
Message sent in past days:
![image](https://user-images.githubusercontent.com/63332774/96566665-7ed9a100-12e3-11eb-949a-dfa5157aa6cf.png)

And also place the timestamp in front of username, like in the image.

Reply:
I agree with you that there should be a new issue for this. I pointed out this because you added `date-fns`. Leave this for now. I will create another issue for this.